plugins/connection/network_cli.py

```diff
@@ -888,7 +910,18 @@ def send(
             response = self.receive(
                 command, prompt, answer, newline, prompt_retry_check, check_all
             )
-            return to_text(response, errors="surrogate_then_replace")
+            response = to_text(response, errors="surrogate_then_replace")
+            # check if the target device is in config mode
+            if self.is_in_config_mode():
```

Comment:
Add check before this condition to see if use_cache is enabled or else it will check this condition for each command